### ISSUE TITLE
feat(model): use deeper MLP classification head instead of single linear layer

### ISSUE THREAD

**Issue by hanneshapke** [dataiku/kiji-proxy#257]
## Summary

Replace the single `nn.Linear` classification heads with small MLPs that include dropout and a non-linear activation, giving the model more capacity to learn complex PII boundary patterns.

## Problem

The current architecture in `model.py` uses the simplest possible classification head — a single linear projection from DistilBERT's hidden states to label logits:

```python
self.pii_classifier = nn.Linear(hidden_size, num_pii_labels)
self.coref_classifier = nn.Linear(hidden_size, num_coref_labels)
```

A single linear layer limits the decision boundary to a hyperplane in the hidden space. PII boundary detection often requires more nuanced distinctions (e.g., distinguishing a name used as part of an email address vs. a standalone name, or recognizing that a sequence of digits is an SSN vs. a zip code based on surrounding context).

## Proposed solution

Replace each single-layer head with a small MLP with dropout:

```python
self.pii_classifier = nn.Sequential(
    nn.Dropout(0.1),
    nn.Linear(hidden_size, hidden_size // 2),
    nn.GELU(),
    nn.Dropout(0.1),
    nn.Linear(hidden_size // 2, num_pii_labels),
)
```

This adds:
- **Pre-projection dropout** for regularization on the encoder output
- **A bottleneck linear layer** (hidden_size → hidden_size // 2) to learn a task-specific representation
- **GELU activation** to introduce a non-linear decision boundary
- **Post-activation dropout** before the final projection
- **Final projection** to label logits

The same pattern should be applied to the coref head.

## Why this helps

- Adds representational capacity without significantly increasing parameter count (one extra `hidden_size × hidden_size/2` matrix per head)
- Dropout layers provide regularization, which is important given the synthetic training data distribution
- Well-established practice for NER — most competitive systems use at least a 2-layer head
- Negligible impact on inference latency

## Notes

- The dropout rate (0.1) and bottleneck ratio (1/2) are reasonable defaults but could be made configurable via `config.py`
- If a CRF layer is added later (#256), the deeper head would produce the emission scores fed into the CRF
